### PR TITLE
feature/limit sell 1000 usd, using Maker price oracle

### PR DESCRIPTION
**Pull request by rmi7**
- Use Maker price oracle to get ETH/USD exchange rate
- rewrote specs to make them readable and comprehensible
- added `eslint` (airbnb) to greatly improve codestyle and get instant error feedback in IDE
- added `eth-gas-reporter` to get gas usage info when running specs (commented out by default, see `truffle.js`
- added 8 specs to test sell limit
- added a Fake price oracle contract (see migrations file) to be used locally and on ropsten (since Maker does not have a deployed contract there)
- added temp KycCertifier contract since we need it to be able to test tier2
- added `isTier1()`, `isTier2()` helper functions in `DetherCore`
- added `CFO` role which has control over the price oracle/sell limit
- added `limitByTier` mapping, to be controlled by CFO

### CODE REVIEW

File: contracts/DetherCore.sol [hunk 1]
```diff
@@ -121,6 +143,10 @@ contract DetherCore is DetherSetup, ERC223ReceivingContract, SafeMath {
     isStarted = true;
   }
 
+  function setPriceOracle (address _priceOracle) external onlyCEO {
```
Comment (Fukunaga42) [inline contracts/DetherCore.sol]: Should be onlyCFO , create this new role in detherAccess

Comment (rmi7) [reply contracts/DetherCore.sol]: what about line 286, should it also be CFO?

```
function setSellDailyLimit(uint _tier, bytes2 _countryId, uint256 _limitUsd) public onlyCSO {
  limitTier[_tier][_countryId] = _limitUsd;
}
```

Comment (Fukunaga42) [reply contracts/DetherCore.sol]: Yes all thing related to the limit should be CFO

**Comment by rmi7** [comment]
ready for review

**Pull request merged**